src/Build/Construction/Solution/SolutionFile.cs

```diff
@@ -202,6 +203,7 @@ internal string FullPath
                 ErrorUtilities.VerifyThrowInternalRooted(value);
                 if (FileUtilities.IsSolutionFilterFilename(value))
                 {
+                    _solutionFilterFile = value;
```

Comment:
nit: I would move the assignment to `ParseSolutionFilter`, just for cosmetic readability reasons. The way it's written now, it makes the reader wonder why the property reads from `_solutionFile` and writes to `_solutionFilterFile`.